src/main/webapp/script.js

```diff
@@ -289,52 +289,57 @@ class AutocompleteDirectionsHandler {
+                // as setting it to 0 will skew data to 0 if passing through an area with very little data. 
+                // So the route score will only be scored based on available data. 
+                let stepAQI;
                 if (!dataGrid[mapRow]) {
-                    console.log("Row map doesnt exist and is undefined.");
+                    console.log("Row map doesnt exist and is undefined. Ignore this step, skip over iteration.");
+                    continue;
                 } else if (!dataGrid[mapRow][mapCol]) {
-                    console.log("Step aqi dosnt exist because mapCol doesn't exist in mapRow. So cant find and set stepAQI");
+                    console.log("Step aqi dosnt exist because mapCol doesn't exist in mapRow. Skip over iteration.");
+                    continue;
                 } else {
                     stepAQI = dataGrid[mapRow][mapCol];
                 }
 
-                console.log(stepAQI);
+                console.log("The Step's AQI is:  " + stepAQI);
                 totalValue += stepWeight * stepAQI;
                 totalWeight += stepWeight;
-                console.log("So far total value is: " + totalValue + " total weight is: "+ totalWeight);
+                console.log("So far total value is: " + totalValue + " total weight is: " + totalWeight);
             }
 
-            // Counting the AQI at the end point of the leg as part of the score. 
-            // Set duration stepWeight as 60 seconds to account of parking upon arrival at end point. 
-            // TODO: Rosanna to rethink on this logic, would it stll apply with waypoints where there are a lot of legs? It can count for stopover time. 
-            let legWeight = 60;
-            let legEndlat = leg["end_location"].lat();
-            let legEndLng = leg["end_location"].lng();
-
-            let endPtRowCol = getGridIndex(legEndlat, legEndLng, aqDataPointsPerDegree);
-            let mapRow = endPtRowCol.row;
-            let mapCol = endPtRowCol.col;
-            let endptAQI = 0;
-
-            // Print to see in console whether mapRow or mapCol is undefined. 
-            if (!dataGrid[mapRow]) {
-                console.log("Row map doesnt exist and is undefined.");
-            } else if (!dataGrid[mapRow][mapCol]) {
-                console.log("Step aqi dosnt exist because mapCol doesn't exist in mapRow. So cant find and set stepAQI");
-            } else {
-                endptAQI = dataGrid[mapRow][mapCol];
-            }
+            if ((legIndex + 1) == legs.length) {
+                // If it is the last leg of the route, count the AQI at the end point of the leg as part of the score. 
+                // Set duration stepWeight as 180 seconds (3 min) to account for parking upon arrival at end point. 
+                let legWeight = 180;
+                let legEndlat = leg["end_location"].lat();
+                let legEndLng = leg["end_location"].lng();
```

Comment:
This duplicates most of the scoring code from above; could you find a way to reuse that code? It would be good to wrap it in a function that can be used here too.